nf_requirements/v1alpha1/capacity.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 Nephio.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +kubebuilder:object:root=true
+type Capacity struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   CapacitySpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status CapacityStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+type CapacitySpec struct {
+	// MaxUplinkThroughput defines the max uplink dataplane throughput
+	MaxUplinkThroughput resource.Quantity `json:"maxUplinkThroughput,omitempty" yaml:"maxUplinkThroughput,omitempty"`
+	// MaxDownlinkThroughput defines the max downlink dataplane throughput
+	MaxDownlinkThroughput resource.Quantity `json:"maxDownlinkThroughput,omitempty" yaml:"maxDownlinkThroughput,omitempty"`
+	// MaxSessions defines the max sessions of the control plane
+	MaxSessions int `json:"maxSessions,omitempty" yaml:"maxSessions,omitempty"`
+	// MaxSubscribers defines the max subscribers
+	MaxSubscribers int `json:"maxSubscribers,omitempty" yaml:"maxSubscribers,omitempty"`
```

Comment:
What's the units here and for sessions? Fine if it's just singular but maybe it should be in 1000s or something?

Reply:
ok I added this in the text